### DISCUSSION TITLE
Extension Instrumentation Class not found

### DISCUSSION THREAD

**Discussion by altaiezior** [open-telemetry/opentelemetry-java-instrumentation#6598] [Q&A] [answered]
Hello I have been trying to additional instrumentation using the extension API but in my Advice method I am not able to use any Singletons.

I tried different mechanisms and I always get class not found in AppLoader

**Comment by laurit** [comment, marked as answer]
Perhaps you forgot to implemented `isHelperClass` in your instrumentation module https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/writing-instrumentation-module.md#tell-the-agent-which-classes-are-a-part-of-the-instrumentation-by-overriding-the-ishelperclass-method

**Reply by altaiezior** [reply]
Thanks for help, yes I was missing this out
